azext_edge/edge/params.py

```diff
@@ -734,3 +734,120 @@ def load_iotops_arguments(self, _):
+            "certificate_reference",
+            options_list=["--certificate-ref", "--cert-ref", "--cr"],
+            help="Reference for the certificate used in authentication. This method of user authentication is not "
+            "supported yet.",
+            arg_group="Authentication",
+        )
+        context.argument(
+            "password_reference",
+            options_list=["--password-ref", "--pr"],
+            help="Reference for the password used in authentication.",
+            arg_group="Authentication"
+        )
+        context.argument(
+            "username_reference",
+            options_list=["--username-reference", "--ur"],
+            help="Reference for the username used in authentication.",
+            arg_group="Authentication"
+        )
+        context.argument(
+            "custom_location_name",
+            options_list=["--custom-location", "--cl"],
+            help="Custom location used to associate asset endpoint with cluster.",
+            arg_group="Associated Resources"
+        )
+        context.argument(
+            "custom_location_resource_group",
+            options_list=["--custom-location-resource-group", "--clrg"],
+            help="Resource group for custom location.",
+            arg_group="Associated Resources"
+        )
+        context.argument(
+            "custom_location_subscription",
+            options_list=["--custom-location-subscription", "--cls"],
+            help="Subscription Id for custom location.",
+            arg_group="Associated Resources"
+        )
+        context.argument(
+            "cluster_name",
+            options_list=["--cluster", "-c"],
+            help="Cluster to associate the asset with.",
+            arg_group="Associated Resources"
+        )
+        context.argument(
+            "cluster_resource_group",
+            options_list=["--cluster-resource-group", "--crg"],
+            help="Resource group for cluster.",
+            arg_group="Associated Resources"
+        )
+        context.argument(
+            "cluster_subscription",
+            options_list=["--cluster-subscription", "--cs"],
+            help="Subscription Id for cluster.",
+            arg_group="Associated Resources"
+        )
+        context.argument(
+            "tags",
+            options_list=["--tags"],
+            help="Asset Endpoint resource tags. Property bag in key-value pairs with the following format: a=b c=d",
+            arg_type=tags_type,
+        )
+
+    with self.argument_context("iot ops asset endpoint certificate") as context:
+        context.argument(
+            "asset_endpoint_profile_name",
+            options_list=["--endpoint"],
+            help="Asset Endpoint name.",
+        )
+        context.argument(
+            "password_reference",
+            options_list=["--password-ref", "--pr"],
+            help="Reference for pem file that contains the certificate password.",
+            arg_group=None
```

Comment:
What happened here where arg_group needs None?

Reply:
I wanted to make sure that this one wouldn't go underneath an arg group (from the parent class) - arg group got inherited even though the param was overridden